worker/task/process_event_delivery.go

```
@@ -211,7 +212,22 @@ func ProcessEventDelivery(endpointRepo datastore.EndpointRepository, eventDelive
 		} else {
 			httpDuration = time.Duration(endpoint.HttpTimeout) * time.Second
 		}
-		resp, err := dispatch.SendWebhook(ctx, targetURL, sig.Payload, project.Config.Signature.Header.String(), header, int64(cfg.MaxResponseSize), eventDelivery.Headers, eventDelivery.IdempotencyKey, httpDuration)
+
+		// Load mTLS client certificate if configured
+		var mtlsCert *tls.Certificate
+		if endpoint.MtlsClientCert != nil {
+			cert, certErr := config.LoadClientCertificate(
+				endpoint.MtlsClientCert.ClientCert,
+				endpoint.MtlsClientCert.ClientKey,
+			)
+			if certErr != nil {
+				log.FromContext(ctx).WithError(certErr).Error("failed to load mTLS client certificate")
+			} else {
+				mtlsCert = cert
```

We should fail fast at this point to avoid needless retries. Wdyt?